### ISSUE TITLE
react/jsx-runtime should be an external

### ISSUE THREAD

**Issue by jrson83** [IgnacioNMiranda/vite-component-library-template#69]
The bundle includes both `react/jsx-runtime`, `production` and `development`.

```sh
vite v4.3.9 building for production...
✓ 17 modules transformed.
dist/style.css                               5.71 kB │ gzip: 1.75 kB
dist/vite-component-library-template.es.js  21.88 kB │ gzip: 6.64 kB
dist/vite-component-library-template.umd.js  14.54 kB │ gzip: 5.77 kB

[vite:dts] Start generate declaration files...
✓ built in 1.86s
[vite:dts] Declaration files built in 1437ms.
```

Both should be excluded from bundle:

```ts
rollupOptions: {
  external: ['react', 'react/jsx-runtime', 'react-dom', 'tailwindcss'],
  output: {
    globals: {
      react: 'React',
      'react/jsx-runtime': 'react/jsx-runtime',
      'react-dom': 'ReactDOM',
      tailwindcss: 'tailwindcss',
    },
  },
},
```

Resulting in:

```sh
vite v4.3.9 building for production...
✓ 9 modules transformed.
dist/style.css                              5.71 kB │ gzip: 1.75 kB
dist/vite-component-library-template.es.js  0.87 kB │ gzip: 0.50 kB
dist/vite-component-library-template.umd.js  1.05 kB │ gzip: 0.62 kB

[vite:dts] Start generate declaration files...
✓ built in 1.63s
[vite:dts] Declaration files built in 1368ms.
 ``

**Comment by IgnacioNMiranda** [comment]
Hey! thanks for opening this and for demonstrating interest in the repo!
Could you explain a bit more why this is not needed in the bundle? Please attach any url, blog post, or news that explain it, as I was not able to find any 🙂🙏🏻

**Comment by jrson83** [comment]
@IgnacioNMiranda as I was looking for examples, how to properly bundle a react component library I found this repo. Since I was wondering, why my setup was including the `jsx-runtime`, I checked the dist output of your repo and it was also including the runtime.

The library package has specified react & react-dom as `peerDependencies`, which includes the JSX runtime. As I understand, a library bundle is meant to be consumed by another project which needs react installed as `dependency`, so it solves the required `peerDependencies`. When the library component is consumed by the project, it receives/uses the JSX runtime from the projects `dependencis`. That's why there is no need to include the runtime in the component library.

```json
"peerDependencies": {
  "react": "18.2.0",
  "react-dom": "18.2.0"
},
```

I found this [related issue](https://github.com/vitejs/vite-plugin-react/issues/17), explaining that the cause is that vite is using the automatic JSX runtime (not classic). And this [stackoverflow thread](https://stackoverflow.com/questions/75234915/why-is-vite-ts-bundling-both-production-and-development-versions-of-react-jsx-ru), providing the same information and fix.

I added the fix, run `build`, then `pnpm pack` in the `vite-component-library-template` root and installed the package as local dependency inside another react project. I used `import { AtButton } from 'vite-component-library-template'` and run `dev/build` without any problems.
